specs/ingress/tests.go

```diff
@@ -58,50 +59,219 @@ func getExternalIP(svc, ns string) (string, error) {
+		// verify kuard installation
+		err = h.CheckDeployment(kuardNs, kuard, 3)
+		gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("deploy/%s in namespace/%s does not have expected replicas", kuard, kuardNs))
+		err = utils.CheckProxyContainer(kuard, kuardNs)
+		gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("could not find proxy container for pods under deploy/%s", kuard))
+	default:
+		utils.TestEmojivotoApp()
+		utils.TestEmojivotoInject()
+	}
+}
 
-	out, stderr, err := h.PipeToLinkerdRun(out, "inject", "-")
-	gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("failed to inject: %s", stderr))
+func testSampleAppUninstall(ingressName string) {
+	h, _ := utils.GetHelperAndConfig()
 
-	_, err = h.KubectlApply(out, utils.NginxNs)
-	gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("failed to apply injected manifests: %s", utils.Err(err)))
+	switch ingressName {
+	case gloo:
+		utils.TestBooksappUninstall()
 
-	err = h.CheckPods(utils.NginxNs, utils.NginxController, 1)
-	gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("failed to verify controller pods: %s", utils.Err(err)))
+	case contour:
+		_, err := h.Kubectl("", "delete", "-f", kuardYAML)
+		gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("could not delete sample application [kuard]: %s", utils.Err(err)))
+	default:
+		utils.TestEmojivotoUninstall()
+	}
+}
 
-	ginkgo.By("Verifying if ingress controller pods have been injected")
-	// Wait upto 3mins for proxy container to show up
-	err = utils.CheckProxyContainer(utils.NginxController, utils.NginxNs)
-	gomega.Expect(err).Should(gomega.BeNil(), utils.Err(err))
+func testControllerInstall(ingressName, controllerYAML, controllerDeployName, controllerNs string) {
+	h, _ := utils.GetHelperAndConfig()
+	ginkgo.By("Setting up ingress controller for Linkerd")
+	installFailMessage := "failed to install ingress controller"
+
+	switch ingressName {
+	case gloo:
+		err := utils.InstallGlooctlBinary()
+		gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("could not install `glooctl` binary: %s", utils.Err(err)))
+
+		_, err = utils.GlooctlRun("install", "gateway")
+		gomega.Expect(err).Should(gomega.BeNil(), utils.Err(err))
+
+	case contour:
+		_, err := h.Kubectl("", "apply", "-f", controllerYAML) // YAML contains inject annotation
+		gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("%s: %s", installFailMessage, utils.Err(err)))
+
+		// verify contour installation
+		err = h.CheckDeployment(controllerNs, controllerNs, 2)
+		gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("deploy/%s in namespace/%s does not have expected replicas", controllerDeployName, controllerNs))
+		err = utils.CheckProxyContainer(controllerDeployName, controllerNs)
+		gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("could not find proxy container for pods under deploy/%s", controllerDeployName))
```

Comment:
> Any reasons why this can't be the same as the default case
In case of `contour`, the number of replicas of the controller deployment is 2, which is different from the default case. Hence, a separate check for that :)